avro-fastserde/src/main/java/com/linkedin/avro/fastserde/SchemaAssistant.java

```diff
@@ -59,15 +59,22 @@
    * the libraries, which define those classes, and put them in the compile classpath.
    */
   private final Set<String> fullyQualifiedClassNameSet = new HashSet<>();
+  /**
+   * Whether this schema assistant is for serializer generation or not.
+   * The reason to differentiate serializer from de-serializer is that for some Avro features, such as java string property,
+   * it is not meaningful for serializer in generic mode.
+   */
+  private final boolean isForSerializer;
 
-  public SchemaAssistant(JCodeModel codeModel, boolean useGenericTypes, Class defaultStringClass) {
+  public SchemaAssistant(JCodeModel codeModel, boolean useGenericTypes, Class defaultStringClass, boolean isForSerializer) {
```

Comment:
can this class (SchemaAssistant) be reused? if so - isnt this hurting reusability? wouldnt it be better to make isForSerializer an argument to method findStringClass()? thats the only place its used right now

Reply:
`SchemaAssistant` is being used in fast serializer/de-serializer  generation, and the main functionality of this class  is to infer the class from schema.
Passing `isForSerializer` will be messy since there  are multiple layers of invocation of 'classFromSchema`, which is using `findStringClass`, and this flag could affect to other data types as well in the future, such as  logical types introduced in 1.8..

Reply:
I think it'd be a somewhat big change to reuse `SchemaAssistant`. At construction time, it gets passed a `JCodeModel` which is stateful and tied to a specific serde class getting generated. If we want to make `SchemaAssistant` stateless, then we would need to add a `JCodeModel` param to most or all `SchemaAssistant` functions, which seems a bit inconvenient.